### PR TITLE
[AgriFood] Updating clients for 2022-11-01-preview API version

### PR DESCRIPTION
**Pull request by bhargav-kansagara**
- Updating clients to support 2022-11-01-preview API version
- Add/Update tests
- Link to swagger - https://github.com/Azure/azure-rest-api-specs/blob/main/specification/agrifood/data-plane/Microsoft.AgFoodPlatform/preview/2022-11-01-preview/agfood.json

### CODE REVIEW

File: eng/Directory.Build.Common.targets
Comment (weshaggard) [inline eng/Directory.Build.Common.targets:32]: Why was this change necessary? We need to keep the forward slashes to ensure we work cross-platform.

File: sdk/agrifood/Azure.Verticals.AgriFood.Farming/src/autorest.md
Comment (weshaggard) [inline sdk/agrifood/Azure.Verticals.AgriFood.Farming/src/autorest.md:8]: This will need to be changed to point to a spec in the public repo.

File: eng/Directory.Build.Common.targets
Comment (bhargav-kansagara) [reply eng/Directory.Build.Common.targets]: Reverted

File: sdk/agrifood/Azure.Verticals.AgriFood.Farming/src/autorest.md
Comment (bhargav-kansagara) [reply sdk/agrifood/Azure.Verticals.AgriFood.Farming/src/autorest.md]: Updated

**Comment by bhargav-kansagara** [comment]
Hi @weshaggard,
Hope you are doing well. Could you please confirm if I can merge this PR?

**Comment by weshaggard** [comment]
@bhargav-kansagara I don't really have any context on this work have you partnered with anyone on the SDK team? @jsquire @pallavit might be better to sign-off or have someone help you.

**Comment by bhargav-kansagara** [comment]
No worries @weshaggard, let me ask Ted.

**Comment by bhargav-kansagara** [comment]
@tg-msft Could you please confirm if I can merge this PR. Major feedback from the review was to have API group specific clients instead of just one FarmBeats client.

**Comment by jsquire** [comment]
@AlexanderSher :  Would you please be so kind as to review and assist with the merge, if needed.

File: sdk/agrifood/Azure.Verticals.AgriFood.Farming/src/Azure.Verticals.AgriFood.Farming.csproj
Comment (AlexanderSher) [inline sdk/agrifood/Azure.Verticals.AgriFood.Farming/src/Azure.Verticals.AgriFood.Farming.csproj:8]: These warning suppressions must be reverted, errors fixed:
- For AZC0001, please create an issue in in azure-sdk repo to add Azure.Verticals to the list of well-known namespaces
- AZC0006 and AZC0007 should not appear for subclient. For the top-level client, they are valid
- For the AZC0012, single-worded type names require approval. We have to provide better names. See see https://github.com/Azure/autorest.csharp/blob/feature/v3/samples/Azure.Analytics.Purview.Account/Customizations.cs  on how customize type names or change the hierarchy of clients if needed (you don't need to specify ParentClient if existing hierarchy works for you).

@tg-msft , @KrzysztofCwalina - do you think it is ok to merge with these warnings suppressed, assuming that they will be addressed before RTM?

Comment (KrzysztofCwalina) [reply sdk/agrifood/Azure.Verticals.AgriFood.Farming/src/Azure.Verticals.AgriFood.Farming.csproj:8]: Shape of the APIs (subclients, naming, etc.) are OK to release in a preview. We just need to fix before GA. 

Package name issues needs to be resolved before we release the first preview. But it seems like this is not the first preview, so in this particular case, we are good to. 

cc: @tg-msft

**Review by jsquire — APPROVED** [review]
Given Krzysztof's comment, approving to unblock

**Pull request merged**
